src/compiler/scala/tools/nsc/typechecker/MethodSynthesis.scala

```diff
@@ -134,28 +134,26 @@ trait MethodSynthesis {
       ImplicitClassWrapper(tree).createAndEnterSymbol()
     }
 
-    def enterGetterSetter(tree: ValDef) {
-      val ValDef(mods, name, _, _) = tree
-      if (nme.isSetterName(name))
-        ValOrValWithSetterSuffixError(tree)
-
-      tree.symbol = (
-        if (mods.isLazy) {
+    // TODO: why is all this symbol creation disconnected from addDerivedTrees,
+    // which creates another list of Field/Getter/Setter factories???
```

Comment:
Looks like this part enters the symbols (which needs to be done before typechecking), and `addDerivedTrees` is called to create the corresponding trees during typechecking.

I agree it would be cleaner to keep track of the factories we create here (maybe as a attachment on the ValDef's symbol?) and avoid the duplicated logic in `allValDefDerived`.